### ISSUE TITLE
plot_top_losses(), object is not subscriptable exception, heatmap 

### ISSUE THREAD

**Issue by omriseemann** [fastai/fastai#2080]
**Describe the bug**

interp.plot_top_losses() throws exception: "object is not subscriptable"
for heatmap=True parameter.

object is an nn.Module object from torch.
with heatmap=False, everything works as expected.

I used learner: vision.Learner

**Provide your installation details**

```text
=== Software === 
python       : 3.7.3
fastai       : 1.0.51
fastprogress : 0.1.21
torch        : 1.0.1.post2
torch cuda   : None / is **Not available** 

=== Hardware === 
No GPUs available 

=== Environment === 
platform     : Linux-4.18.0-18-generic-x86_64-with-debian-buster-sid
distro       : #19-Ubuntu SMP Tue Apr 2 18:13:16 UTC 2019
conda env    : fastai
.....
```

**To Reproduce**

try: 
```python
import fastai.vision as faiv
import torch
from torchvision import models
import torch.nn as nn
import torch.nn.functional as F
import torchvision
import torchvision.transforms as transforms

model_alexnet = models.alexnet()
model_alexnet.classifier[6] = nn.Linear(4096, 3) #change 3 to number of output labels
data = '' # add some data object here
learn1 = faiv.Learner(data, model_alexnet, metrics=faiv.accuracy)
learn1.fit_one_cycle(1,1e-3)
interp = faiv.ClassificationInterpretation.from_learner(learn1)
interp.plot_top_losses(2,heatmap=False)
interp.plot_top_losses(2,heatmap=True)
```

**Expected behavior**

**Screenshots**

**Additional context**

look at: 
https://github.com/fastai/fastai/blob/master/fastai/vision/learner.py

function: 
_cl_int_plot_top_losses

line: 148

**Comment by sgugger** [comment]
With `heatmap=True`, the model is expected to come from `cnn_learner`, so you should create it this way: 
```
learn = cnn_learner(data, models.alexnet)
```

**Comment by omriseemann** [comment]
I think outing a functionality in Learner and expecting it to come from
cnn_learner is an issue..
Your discretion

**Comment by sgugger** [comment]
I didn't write the heatmap functionality and I only discovered it wasn't working on all models with your issue. Will make the default be `False`. If you want to make a PR to add a warning in the docs it will only work with a `cnn_learner` it would be much appreciated.

**Comment by sgugger** [comment]
Made a few changes: `heatmap` will default to `True` if it detects the model is like a `cnn_learner` produced model, `False` otherwise, and if you try to use `heatmap=True` with a model that doesn't have the right characteristics, it will print a useful error message.

**Comment by sebastienwood** [comment]
Had this issue today on latest version, while I have not a cnn_learner I didn't specify the heatmap arg. It however produced an error.

```
model = models.WideResNet(num_groups=3, N=3,num_classes=100,k=6,drop_p=args.droprate)
learn = Learner(data, model, loss_func=nn.CrossEntropyLoss(), metrics=[accuracy])
learn.lr_find()
mingradlr = learn.recorder.min_grad_lr
learn.fit_one_cycle(5, mingradlr)
interp.plot_top_losses(9)
```

The error message : 
```
TypeError                                 Traceback (most recent call last)
<ipython-input-12-776eb7830b8b> in <module>
----> 1 interp.plot_top_losses(9)

~/anaconda3/lib/python3.7/site-packages/fastai/vision/learner.py in _cl_int_plot_top_losses(self, k, largest, figsize, heatmap, heatmap_thresh, return_fig)
    145             xb,_ = self.data.one_item(im, detach=False, denorm=False)
    146             m = self.learn.model.eval()
--> 147             with hook_output(m[0]) as hook_a:
    148                 with hook_output(m[0], grad= True) as hook_g:
    149                     preds = m(xb)

TypeError: 'WideResNet' object is not subscriptable
```

It can be resolved by specifing the heatmap arg to False when calling plot_top_losses.

**Comment by sgugger** [comment]
The bug fix isn't in the latest version (v1.0.52 was released before it) so you have to use an editable install. Just tested your example and it worked fine on master, without the need to pass `heatmap=False`.

**Comment by sebastienwood** [comment]
Ok my bad ! :)
